lens/src/mappings/index.ts

```diff
@@ -0,0 +1,187 @@
+    userToProfileMap.hashes = [txHash]
+  } else {
+    log.error('one User multiple profile,txHash {}', [txHash])
+    let profileArr = userToProfileMap.profile
+    if (profileArr == null) {
+      throw new Error('')
+    } else {
+      profileArr.push(profileId)
+    }
+    userToProfileMap.profile = profileArr
+    let lensNameArr = userToProfileMap.lensNames
+    if (lensNameArr == null) {
+      throw new Error('')
+    } else {
+      lensNameArr.push(event.params.handle)
+    }
+    userToProfileMap.lensNames = lensNameArr
+    let hashArr = userToProfileMap.hashes
+    if (hashArr == null) {
+      throw new Error('')
+    } else {
+      hashArr.push(txHash)
+    }
+    userToProfileMap.hashes = hashArr
+    userToProfileMap.count = userToProfileMap.count.plus(BIGINT_ONE)
+  }
+  userToProfileMap.save()
+  airstack.social.trackSocialUserAndProfileRegisteredTransaction(
+    event.block,
+    txHash,
+    event.logIndex,
+    ADDRESS_ZERO,
+    to.toHexString(),
+    profileId.toString(),
+    LENSHUB_ADDRESS.toHexString(),
+    to.toHexString(),
+    new Array<airstack.social.AirExtraData>(),
+    event.params.handle,
+    new Array<airstack.social.AirExtraData>(),
+    BIG_INT_ZERO,
+  )
+}
+export function handleTransfer(event: Transfer): void {
+  let txHash = event.transaction.hash
+  let transferEntity = TransferEntity.load(event.params.tokenId.toString())
+  if (transferEntity == null) {
+    transferEntity = new TransferEntity(event.params.tokenId.toString())
+    transferEntity.hashes = [txHash.toHexString()]
+    transferEntity.fromTo = [
+      event.params.from
+        .toHexString()
+        .concat('-')
+        .concat(event.params.to.toHexString()),
+    ]
+    transferEntity.tokenId = event.params.tokenId
+    transferEntity.count = BIGINT_ONE
+  } else {
+    let hashArr = transferEntity.hashes
+    if (hashArr == null) {
+      throw new Error('')
+    } else {
+      hashArr.push(txHash.toHexString())
+    }
+    transferEntity.hashes = hashArr
+    let fromTo = transferEntity.fromTo
+    if (fromTo == null) {
+      throw new Error('')
+    } else {
+      fromTo.push(
+        event.params.from
+          .toHexString()
+          .concat('-')
+          .concat(event.params.to.toHexString()),
+      )
+    }
+    transferEntity.fromTo = fromTo
+    transferEntity.count = transferEntity.count.plus(BIGINT_ONE)
+  }
+  transferEntity.save()
+  if (event.params.from.toHexString() != ADDRESS_ZERO) {
+    // changing default Profile if this profileId is previous default
+    let profileId = getDefaultProfileIdfromUser(event.params.from)
+    if (profileId == event.params.tokenId) {
+      log.debug('default profile reset ,hash {}', [txHash.toHexString()])
+      airstack.social.trackSocialUserDefaultProfileChange(
+        event.block,
+        txHash.toHexString(),
+        event.logIndex,
+        event.transaction.from.toHexString(),
+        event.address.toHexString(),
+        '',
```

Comment:
not sure if we should be passing empty strings for a tokenId, should be a seperate bool flag which can be used to check the condition.